### PR TITLE
Add Cloudauth Account resource Tests

### PR DESCRIPTION
**Pull request by ravinadhruve10**
Change summary:
----------------
1. Added tests for new cloudauth based CloudAccount resource.
2. Minor refactoring of marshal/unmarshal methods of cloudauth client.
3. Added the tests for the marshal/unmarshal methods.

### CODE REVIEW

**Review by mdkulkarni15 — APPROVED** [review]

File: sysdig/resource_sysdig_secure_cloud_auth_account_test.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	return fmt.Sprintf(`
+resource "sysdig_secure_cloud_auth_account" "sample" {
+  provider_id   = "sample-%s"
+  provider_type = "PROVIDER_GCP"
+  enabled       = "true"
+}
+`, accountID)
+}
+
+func secureCloudAuthAccountMinimumConfiguration(accountID string) string {
```
Comment (mdkulkarni15) [inline sysdig/resource_sysdig_secure_cloud_auth_account_test.go:58]: What's the different between this one and secureCloudAuthAccountWithID?

Comment (ravinadhruve10) [reply sysdig/resource_sysdig_secure_cloud_auth_account_test.go:58]: Yeah, it can be a single test. I'll push a fix.

Comment (ravinadhruve10) [reply sysdig/resource_sysdig_secure_cloud_auth_account_test.go:45]: Done.

File: sysdig/resource_sysdig_secure_cloud_auth_account_test.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+			{
+				ResourceName:      "sysdig_secure_cloud_auth_account.sample-1",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func secureCloudAuthAccountWithFC(accountID string) string {
```
Comment (mdkulkarni15) [inline sysdig/resource_sysdig_secure_cloud_auth_account_test.go:81]: can this be part of  TestAccSecureCloudAuthAccount itself? you can run secureCloudAuthAccountWithFC and secureCloudAuthAccountMinimumConfiguration in parallel right?

**Review by mdkulkarni15 — APPROVED** [review]

File: sysdig/resource_sysdig_secure_cloud_auth_account_test.go (hunk 2 above)
Comment (ravinadhruve10) [reply sysdig/resource_sysdig_secure_cloud_auth_account_test.go:81]: So from the confluence page on TF resources, it is mentioned that each step under the same test modifies the configuration of the same resource that gets created. After all steps are executed, terraform will destroy the created resource. Hence I separated them into different tests. First Test does basic sanity, the second Test tests full payload.

**Review by cgeers — APPROVED** [review]

**Pull request merged**